src/Raven.Client/Documents/Session/GroupByDocumentQuery.cs

```diff
@@ -45,6 +46,15 @@ public IDocumentQuery<T> SelectCount(string projectedName = null)
             _query.GroupByCount(projectedName);
             return _query;
         }
+        
+        public IGroupByDocumentQuery<T> Filter(Action<IFilterFactory<T>> builder)
+        {
+            _query.TurnOnFilter();
```

Comment:
I wonder if instead of doing `TurnOnFilter();` / `TurnOffFilter();` shouldn't we introduce some scope like:

```
using (_query.FilterMode())
{
}
````

Reply:
Would be my preference as well

Reply:
Done